### PR TITLE
ENH: Add option to define reference view from which the VR view is in…

### PR DESCRIPTION
**Pull request by cpinter**
…itialized

### CODE REVIEW

**Comment by cpinter** [comment]
I haven't been able to test this yet, but it's quite straightforward. I can test it tomorrow morning.

File: VirtualReality/Widgets/qMRMLVirtualRealityView.cxx [hunk 1]
```diff
@@ -221,6 +222,33 @@ void qMRMLVirtualRealityViewPrivate::destroyRenderWindow()
   this->Lights = NULL;
 }
 
+//---------------------------------------------------------------------------
+void qMRMLVirtualRealityViewPrivate::initializeViewFromReferenceView()
+{
+  qMRMLThreeDWidget* referenceWidget = NULL;
+  qMRMLLayoutManager* layoutManager = qSlicerApplication::application()->layoutManager();
+  if ( !this->MRMLVirtualRealityViewNode || !this->MRMLVirtualRealityViewNode->GetReferenceViewNode()
+    || !vtkMRMLViewNode::SafeDownCast(this->MRMLVirtualRealityViewNode->GetReferenceViewNode()) )
```
Comment (jcfr) [inline VirtualReality/Widgets/qMRMLVirtualRealityView.cxx]: Nitpick: Could probably add a function named `hasReferenceViewNode()`

**Comment by jcfr** [comment]
I checked that it compiled :+1:

**Comment by cpinter** [comment]
Thanks!

I just added a referenceViewNode function but as it's merged, I'd rather not go through another PR so reverted. I'll do it next time or if there are any problems :)

**Comment by lassoan** [comment]
We should not store historical information in MRML nodes (how we get to the current state), only the current state. So, the "VR view camera is initialized from the camera of the reference view" should not be stored. However, I think a reference view node is useful as we could use that when we implement features, such as:
- content sync: show/hide nodes in the VR view that are shown in the reference view
- camera reset: position what is in front of the VR view to be the same as the reference view shows
- camera sync: update the reference view's camera to show what you see in the VR view

If you agree, I'll update documentation of Get/SetReferenceViewNode methods to describe this continuous relationship to the reference view.

**Comment by cpinter** [comment]
I agree that the phrasing of the comment is not ideal. Actually, whenever you set a reference node, this synchronization of the camera is done (in updateWidgetFromMRML).

**Comment by lassoan** [comment]
Probably we should separate the two operations (so that developers can choose to sync VR->reference or reference->VR). We could add logic methods for various reset and one time/continuous synchronization operations.

**Comment by cpinter** [comment]
Sounds good. If we don't mention the camera synchronization in the doc (it'd be still nice to add the points you listed), and don't do it in updateWidgetFromMRML, but have a synchronize camera function, then it makes better sense.

**Pull request merged**